locals.tf

```diff
@@ -0,0 +1,144 @@
+# -------------------------------------------------------------------------------------------------
+# Policy transformations
+# -------------------------------------------------------------------------------------------------
+
+locals {
+  # Transforn from:
+  #
+  # policies = [
+  #   {
+  #     name = ""
+  #     path = ""
+  #     desc = ""
+  #     file = ""
+  #   },
+  #   {
+  #     name = ""
+  #     path = ""
+  #     desc = ""
+  #     file = ""
+  #   },
+  # }
+  #
+  # Into the following format:
+  #
+  # policies = {
+  #   "<policy-name>" = {
+  #     name = ""
+  #     path = ""
+  #     desc = ""
+  #     file = ""
+  #   }
+  #   "<policy-name>" = {
+  #     name = ""
+  #     path = ""
+  #     desc = ""
+  #     file = ""
+  #   }
+  # }
+  policies = { for i, v in var.policies : var.policies[i]["name"] => v }
+}
+
+
+# -------------------------------------------------------------------------------------------------
+# Role Policy transformations
+# -------------------------------------------------------------------------------------------------
+
+locals {
+
+  # This local combines var.roles and var.policies and creates its own list as shown below:
+  #
+  # role_policies = [
+  #   {
+  #     "<role-name>:<policy-name>" = {
+  #        "name" = "<policy-name>"
+  #        "path" = "<policy-path>"
+  #        "desc" = "<policy-desc>"
+  #        "file" = "<policy-file>"
+  #     }
+  #   },
+  # ]
+
+  _role_policies = flatten([
+    for i, role in var.roles : [
+      for j, policy in lookup(var.roles[i], "policies", {}) : {
+        "${var.roles[i]["name"]}:${var.roles[i]["policies"][j]}" = local.policies[var.roles[i]["policies"][j]]
+      }
+    ]
+  ])
+  # The fix to bring it into the format stated at the top of this file
+  role_policies = {
+    for i, v in local._role_policies :
+    keys(local._role_policies[i])[0] => local._role_policies[i][keys(local._role_policies[i])[0]]
+  }
+}
+
```

Comment:
i would simplify this structure a bit like so 

```
  rp = flatten([
    for role in var.roles : [
      for policy in role["policies"] : {
        role_name   = role.name
        policy_name = policy
        policy      = local.policies[policy]
      }
    ]
  ])

  role_policies = { for obj in local.rp : "${obj.role_name}:${obj.policy_name}" => obj.policy }
```

Reply:
done